sql/distsql: TestAggregatorAgainstProcessor failed

[(sql/distsql).TestAggregatorAgainstProcessor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2198457&tab=buildLog) on [master@2022bd0431f62b4b73c5377cac638bfa61bfb01d](https://github.com/cockroachdb/cockroach/commits/2022bd0431f62b4b73c5377cac638bfa61bfb01d):

Fatal error:
```
panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple [recovered]
	panic: expected *DGeometry, found *tree.DTuple

```

Stack:
```
goroutine 55 [running]:
testing.tRunner.func1(0xc0004dcb00)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x432d2e0, 0xc0007d3500)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:103 +0x22f
panic(0x432d2e0, 0xc0007d3500)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDGeometry(0x56a5040, 0xc0007b6de0, 0xc0007b6de0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:2859 +0xd8
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*stExtentAgg).Add(0xc00010d2e8, 0x56874c0, 0xc00010a010, 0x57d3f40, 0xc0007b6de0, 0x0, 0x0, 0x0, 0xc00055a6f8, 0x1a537f8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:661 +0x6a
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket(0xc0013a2c00, 0xc000907560, 0x6, 0x6, 0x0, 0x0, 0x0, 0xc00127b630, 0x5, 0x5, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:814 +0x3e7
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRow(0xc0013a2c00, 0xc000907560, 0x6, 0x6, 0xc000907560, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:875 +0xbe
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).accumulateRows(0xc0013a2c00, 0x4, 0xc00055a940, 0x1d7305d, 0xc000146c00, 0xc00055a940)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:537 +0x222
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*orderedAggregator).Next(0xc0013a2c00, 0x4c62320, 0xc000146c00, 0x56874c0, 0xc00010a010)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:732 +0x14c
github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator(0x0, 0x0, 0x0, 0x0, 0xc0009f6b20, 0x1, 0x1, 0xc0009f6b40, 0x1, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:197 +0xfbf
github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor(0xc0004dcb00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:248 +0x1266
testing.tRunner(0xc0004dcb00, 0x4c62420)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestAggregatorAgainstProcessor
--- FAIL: TestAggregatorAgainstProcessor (0.09s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestAggregatorAgainstProcessor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/distsql/columnar_operators_test.go
+++ b/pkg/sql/distsql/columnar_operators_test.go
@@ -164,7 +164,7 @@ func TestAggregatorAgainstProcessor(t *testing.T) {
 						for j := range aggFnInputTypes {
 							aggFnInputTypes[j] = sqlbase.RandType(rng)
 						}
-						// There is a special case for concat_agg, string_agg,
+						// There is a special case for concat_agg, string_agg, st_extent
 						// and st_makeline when at least one argument is a
 						// tuple. Such cases pass GetAggregateInfo check below,
 						// but they are actually invalid, and during normal
@@ -175,7 +175,8 @@ func TestAggregatorAgainstProcessor(t *testing.T) {
 						switch aggFn {
 						case execinfrapb.AggregatorSpec_CONCAT_AGG,
 							execinfrapb.AggregatorSpec_STRING_AGG,
-							execinfrapb.AggregatorSpec_ST_MAKELINE:
+							execinfrapb.AggregatorSpec_ST_MAKELINE,
+							execinfrapb.AggregatorSpec_ST_EXTENT:
 							for _, typ := range aggFnInputTypes {
 								if typ.Family() == types.TupleFamily {
 									invalid = true

